pkg/runtime/k0s/utils.go

```diff
@@ -99,3 +55,82 @@ func (k *Runtime) WaitK0sReady(ssh ssh.Interface, host net.IP) error {
 	}
 	return errors.New("failed to start k0s: failed to get k0s status after 10 seconds")
 }
+
+func GetClientFromConfig(adminConfPath string) (runtimeClient.Client, error) {
+	adminConfig, err := clientcmd.BuildConfigFromFlags("", adminConfPath)
+	if nil != err {
+		return nil, err
+	}
+
+	var ret runtimeClient.Client
+
+	timeout := time.Second * 30
+	err = wait.PollImmediate(time.Second*10, timeout, func() (done bool, err error) {
+		cli, err := runtimeClient.New(adminConfig, runtimeClient.Options{})
+		if nil != err {
+			return false, err
+		}
+
+		ns := corev1.Namespace{}
+		if err := cli.Get(context.Background(), runtimeClient.ObjectKey{Name: "default"}, &ns); nil != err {
+			return false, err
+		}
+
+		ret = cli
+
+		return true, nil
+	})
+
+	return ret, err
+}
+
+func (k *Runtime) WaitSSHReady(tryTimes int, hosts ...net.IP) error {
+	eg, _ := errgroup.WithContext(context.Background())
+	for _, h := range hosts {
+		host := h
+		eg.Go(func() error {
+			for i := 0; i < tryTimes; i++ {
+				if err := k.infra.Ping(host); err == nil {
+					return nil
+				}
+				time.Sleep(time.Duration(i) * time.Second)
+			}
+			return fmt.Errorf("wait for [%s] ssh ready timeout, ensure that the IP address or password is correct", host)
+		})
+	}
+	return eg.Wait()
+}
+
+func (k *Runtime) getNodeName(host net.IP) (string, error) {
+	client, err := k8s.NewK8sClient()
```

Comment:
at k0s, is this `NewK8sClient` work ? Or we can use k0s driver to get node name. Or ,it just a utils func also used by others runtime?

Reply:
Yep, after I fetch k0s /var/lib/k0s/pki/admin.conf to /root/.kube/config, it also effctive.